### PR TITLE
Do not zoom for longPress add waypoint, zoom to mission in all other cases.

### PR DESCRIPTION
**Pull request by btgalush**

### CODE REVIEW

File: iGCS/UserInterface/ViewControllers/WaypointsViewController.m [hunk 1]
```diff
@@ -141,14 +141,18 @@ - (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interface
...
 }
 
-- (void) resetWaypoints {
-    [self replaceMission:self.waypoints];
+- (void) resetWaypointsZoomToMission:(BOOL)zoomToMission {
+    [self replaceMission:self.waypoints zoomToMission:zoomToMission];
 }
 
 - (void) replaceMission:(WaypointsHolder*)mission {
+    [self replaceMission:mission zoomToMission:YES];
```
Comment (cnatoli) [inline iGCS/UserInterface/ViewControllers/WaypointsViewController.m]: Duplicate method introduced to base class. Remove this one.

File: iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m [hunk 2]
```diff
@@ -143,8 +143,12 @@ - (void) configureRoutePolylines:(CLLocationCoordinate2D*)coords withLength:(NSU
 
 // 1e-5 is very approximately 1m
 #define CLLocationCoordinate2DEqual(x, y) (fabs((x).latitude - (y).latitude) < 1e-5 && fabs((x).longitude - (y).longitude) < 1e-5)
+- (void) replaceMission:(WaypointsHolder*)mission
```
Comment (cnatoli) [inline iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m]: Add spacing between top level forms.

Might also want to move the #define, as it is associated with the method below this one (not that either location is ideal)

File: iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m [hunk 3]
```diff
@@ -143,8 +143,12 @@ - (void) configureRoutePolylines:(CLLocationCoordinate2D*)coords withLength:(NSU
...
+{
```
Comment (cnatoli) [inline iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m]: Brace style inconsistent.

File: iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m [hunk 4]
```diff
@@ -176,7 +180,9 @@ - (void) replaceMission:(WaypointsHolder*)mission {
     // Set the map extents
     NSUInteger offset = isHomeWPInitialised ? 0 : 1;
     if (offset < numWaypoints) {
-        [self zoomInOnCoordinates:coords+offset withLength:numWaypoints-offset]; // don't include HOME/0 in zoom area when not initialised
+        if(zoomToMission) {
```
Comment (cnatoli) [inline iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m]: Space between forms please.

File: iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m [hunk 5]
```diff
@@ -176,7 +180,9 @@ - (void) replaceMission:(WaypointsHolder*)mission {
     // Set the map extents
     NSUInteger offset = isHomeWPInitialised ? 0 : 1;
     if (offset < numWaypoints) {
```
Comment (cnatoli) [inline iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m]: Merge two if statements, or swap ordering and put the declaration of offset in the outer if.

Comment (btgalush) [inline iGCS/UserInterface/ViewControllers/WaypointMapBaseController.m]: Comments addressed.

**Comment by cnatoli** [comment]
+1

**Pull request merged**